cmake/configure.cmake

```diff
@@ -68,6 +68,25 @@ else()
     if(NOT CUDNN_FOUND)
         message(FATAL_ERROR "Paddle need cudnn to compile")
     endif()
+    if (NOT NCCL_INCLUDE_DIR)
+        if (APPLE)
+            message(FATAL_ERROR "Paddle need nccl header to compile GPU version. "
+                    "If you do not need MultiGPU support on macos, "
+                    "you can just grab the nccl.h to search path, "
+                    "and enable dso support."
+                    "Even though there is no offical MacOS version of nccl, "
+                    "you can still compile nccl for MacOS by "
+                    "https://github.com/NVIDIA/nccl/pull/62")
+        else()
+            message(FATAL_ERROR "Paddle need nccl header to compile")
+        endif()
+    endif()
+    if (NOT WITH_DSO)
```

Comment:
It seems that we can marge the two conditions on L84 and L85 into one

```cmake
if (NOT WITH_DSO AND NOT NCCL_LIBRARY)
  message(FATAL_ERROR "Paddle needs the NCCL libraries with WITH_DSO=OFF")
endif()
```

Reply:
Done.